catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/WhenNfcBrickTest.java

```diff
@@ -0,0 +1,488 @@
+	private static final String NFC_NDEF_STRING_1 = "I am the first!";
+	private static final String NFC_NDEF_STRING_2 = "And I am the second!";
+
+	private static final String READ_TAG_ID = "readTagId";
+	private static final String READ_TAG_MESSAGE = "readTagMessage";
+	private static final String NUM_DETECTED_TAGS = "numDetectedTags";
+
+	private static final String TAG_ID_1 = NfcHandler.byteArrayToHex(FIRST_TEST_TAG_ID.getBytes());
+	private static final String TAG_ID_2 = NfcHandler.byteArrayToHex(SECOND_TEST_TAG_ID.getBytes());
+
+	private UserVariable readTagId;
+	private UserVariable readTagMessage;
+	private UserVariable numDetectedTags;
+
+	private static List<NfcTagData> tagDataList;
+
+	@Rule
+	public BaseActivityInstrumentationRule<ScriptActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(ScriptActivity.class, true, false);
+	private int nfcBrickPosition;
+	private int setVariableIDPosition;
+	private int setVariableMessagePosition;
+
+	private Script createProjectWithNfcAndSetVariable(String projectName) {
+		Project project = new Project(null, projectName);
+
+		DataContainer dataContainer = project.getDefaultScene().getDataContainer();
+
+		readTagId = dataContainer.addProjectUserVariable(READ_TAG_ID);
+		readTagMessage = dataContainer.addProjectUserVariable(READ_TAG_MESSAGE);
+		numDetectedTags = dataContainer.addProjectUserVariable(NUM_DETECTED_TAGS);
+
+		Sprite sprite = new Sprite("testSprite");
+		Script script = new WhenNfcScript();
+
+		SetVariableBrick setVariableBrickId = new SetVariableBrick(Sensors.NFC_TAG_ID);
+		setVariableBrickId.setUserVariable(readTagId);
+		script.addBrick(setVariableBrickId);
+
+		SetVariableBrick setVariableBrickMessage = new SetVariableBrick(Sensors.NFC_TAG_MESSAGE);
+		setVariableBrickMessage.setUserVariable(readTagMessage);
+		script.addBrick(setVariableBrickMessage);
+
+		ChangeVariableBrick changeVariableBrickNumDetectedTags = new ChangeVariableBrick(new Formula(1), numDetectedTags);
+		script.addBrick(changeVariableBrickNumDetectedTags);
+
+		sprite.addScript(script);
+		project.getDefaultScene().addSprite(sprite);
+		ProjectManager.getInstance().setProject(project);
+		ProjectManager.getInstance().setCurrentSprite(sprite);
+
+		tagDataList = ProjectManager.getInstance().getCurrentSprite().getNfcTagList();
+		NfcTagData tagData = new NfcTagData();
+		tagData.setNfcTagName(getResourcesString(R.string.test_tag_name_1));
+		tagData.setNfcTagUid(NfcHandler.byteArrayToHex(FIRST_TEST_TAG_ID.getBytes()));
+		tagDataList.add(tagData);
+
+		NfcTagData tagData2 = new NfcTagData();
+		tagData2.setNfcTagName(getResourcesString(R.string.test_tag_name_2));
+		tagData2.setNfcTagUid(NfcHandler.byteArrayToHex(SECOND_TEST_TAG_ID.getBytes()));
+		tagDataList.add(tagData2);
+
+		return script;
+	}
+
+	@Before
+	public void setUp() throws Exception {
+		createProjectWithNfcAndSetVariable(PROJECT_NAME);
+		numDetectedTags.setValue(0);
+
+		nfcBrickPosition = 0;
```

Comment:
set those in createProject method right next to the initializations of the bricks / or where you add the bricks to the scriptList, so it would become more clear what those are good for.